platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -341,18 +341,18 @@ mender_storage_get_artifact_name(char **artifact_name) {
 }
 
 mender_err_t
-mender_storage_save_update_state(mender_update_state_t state, const char *artifact_type) {
+mender_storage_save_update_state(const mender_update_state_t state, const char *artifact_type) {
     assert(NULL != artifact_type);
 
     size_t artifact_type_len;
 
-    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, state, sizeof(state))) {
+    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, (const void *)state, sizeof(state))) {
         mender_log_error("Unable to save update state");
         return MENDER_FAIL;
     }
 
-    artifact_type_size = strlen(artifact_type) + 1;
-    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_ARTIFACT_TYPE, artifact_type, artifact_type_size)) {
+    artifact_type_len = strlen(artifact_type) + 1;
+    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_ARTIFACT_TYPE, artifact_type, artifact_type_len)) {
```

Comment:
It's not a length with the `+1`, that's why I preferred the word *size*

Reply:
:ok_hand: - I will modify the declaration instead, then.